aslam_cv/aslam_time/include/aslam/implementation/Duration.hpp

```diff
@@ -165,7 +166,7 @@ boost::posix_time::time_duration DurationBase<T>::toBoost() const {
 #if defined(BOOST_DATE_TIME_HAS_NANOSECONDS)
   return bt::seconds(sec) + bt::nanoseconds(nsec);
 #else
-  return bt::seconds(sec) + bt::microseconds(nsec / 1000.0);
+  return bt::seconds(sec) + bt::microseconds(static_cast<int32_t>(round(nsec / 1000.0)));
```

Comment:
Looks like previous behavior was to truncate, not round (see https://github.com/boostorg/date_time/blob/dcab4d2bfaf68138f1d3ee80ae6ae205f72c0579/include/boost/date_time/time_duration.hpp#L281), so maybe it would be good to keep that. I.e. just static cast to int (which was the previous interface), or simply do integer division in the first place.

Reply:
But integer division will yield 0 if the input is not dividable, so I think a static cast is the best option, right?

Reply:
I'm not sure what you mean. I think both should give the same result (truncation).

Reply:
But yes, LGTM now.